### PR TITLE
Test build GDNative library in CI

### PR DESCRIPTION
**Pull request by o01eg**
Uses test class from this `README.md` and `SConstruct` from "Compiling the plugin" paragraph of Godot's GDNative example.

### CODE REVIEW

**Comment by Calinou** [comment]
Shoudn't this pull request be made on the `master` branch instead of the `3.2` branch? We usually merge PRs against the `master` branch and cherry-pick improvements selectively to stable branches.

**Comment by o01eg** [comment]
@Calinou I could try it but next PR will execute actual godot 3.2.3 to test if plugin works.

**Comment by bruvzg** [comment]
In case of this repo `master` branch is what should be used for 3.2.3, `3.2` branch is just an outdated version.

**Comment by o01eg** [comment]
@bruvzg Are there any documentations about branches and tags and their appropriate Godot engine versions?

**Comment by bruvzg** [comment]
> Are there any documentations about branches and tags and their appropriate Godot engine versions?

I do not think there's any.

But, at this point, no work for Godot `4.0`/`master` support was done (is's GDNative C API is subject to change).

Also, `godot_headers` repo is outdated as well, it's best to use headers directly for the engine build (`modules/gdnative`). I'm not sure why there's a separate repo for headers at all, it's just a misleading distraction.

**Comment by o01eg** [comment]
Wonder why windows unable to find `cl` second time.

**Comment by o01eg** [comment]
I have tried to copy code from working `SConstrcut` without success so for Windows I currently skip building plugin.

**Comment by Calinou** [comment]
> Also, godot_headers repo is outdated as well, it's best to use headers directly for the engine build (modules/gdnative). I'm not sure why there's a separate repo for headers at all, it's just a misleading distraction.

I think this was done so that third parties can use `godot_headers` as a submodule. Otherwise, you'd have to add the entire Godot repository as a submodule, which is much larger.

That said, we can probably find a different way to do this. For instance, we could add a ZIP of the Godot GDNative headers matching a given release on GitHub Releases every time a Godot release is tagged.

**Comment by o01eg** [comment]
Why not make Godot itself use `godot_headers` as a submodule?

**Comment by Calinou** [comment]
> Why not make Godot itself use `godot_headers` as a submodule?

For UX reasons, we don't want to use submodules in the main Godot repository. (Lots of people forget to initialize submodules then notice that compiling doesn't work.)

**Comment by Calinou** [comment]
I'm switching to GitHub Actions in this PR: https://github.com/godotengine/godot-cpp/pull/486

This can be ported to GitHub Actions once the PR above is merged.

**Comment by o01eg** [comment]
Nice, I'll rebase it and following PR then.

**Comment by Zylann** [comment]
We switched to Github actions so this part should be updated using it

**Comment by o01eg** [comment]
Rebased and updated to Github Actions

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -125,6 +129,10 @@ jobs:
           path: bin/libgodot-cpp.osx.release.64.a
           if-no-files-found: error
 
+      - name: Build test GDNative library
+        run: |
+          scons target=release platform=osx bits=64 -j $(sysctl -n hw.logicalcpu) -C test;
```
Comment (Zylann) [inline .github/workflows/ci.yml:134]: Is `;` necessary?

Comment (o01eg) [reply .github/workflows/ci.yml:134]: No, I just copied it from previous travis yml

Comment (o01eg) [reply .github/workflows/ci.yml]: Fixed.

**Comment by vnen** [comment]
Superseded by #477

**Pull request closed without merging**